ui/perfherder/helpers.js

```diff
@@ -429,6 +429,149 @@ export const addResultsLink = (taskId) => {
   return `${taskLink}${taskId}${resultsPath}`;
 };
 
+export class TextualSummary {
+  constructor(alerts, alertSummary, copySummary = null, alertsWithVideos = []) {
+    this.alerts = alerts;
+    this.alertSummary = alertSummary;
+    this.copySummary = copySummary;
+    this.alertsWithVideos = alertsWithVideos;
+    this.resultStr = '';
+  }
+
+  get markdown() {
+    const improved = sortBy(
+      this.alerts.filter((alert) => !alert.is_regression),
+      'amount_pct',
+    ).reverse();
+    const regressed = sortBy(
+      this.alerts.filter(
+        (alert) =>
+          alert.is_regression && alert.status !== alertStatusMap.invalid,
+      ),
+      'amount_pct',
+    ).reverse();
+
+    // add summary header if getting text for clipboard only
+    if (this.copySummary) {
```

Comment:
I think we could extract these 2 clipboard related conditionals in a separate function.